### DISCUSSION TITLE
img2PDF：好用的圖片轉PDF檔函式庫

### DISCUSSION THREAD

**Discussion by LeoTsai999** [1101-GS4719-C/Python-Community-GS4719-C#2107] [優秀Python套件]
img2PDF是一個能夠將圖片轉換為PDF檔案的函式庫。我曾經使用它將中央大學eeclass中，教授未開放下載僅開放檢視的簡報檔，利用開發者工具下載後，將其合併為一個PDF檔。我認為學習這個函式庫的好處是可以為我們省下向adobe公司付費獲得軟體授權的費用。畢竟這類PDF工具，大多需要向adobe公司付費訂閱軟體或使用網路上安全性不明的線上轉換工具。
以下為我所設計的一個使用範例：
```python
import os
import img2pdf

# 列出目錄中的所有jpg文件
jpg_files = [file for file in os.listdir('.') if file.endswith('.jpg')]

# 對文件進行排序以確保一致的順序
jpg_files.sort(key=lambda x: int(x.split('.')[0]))

# 將所有 jpg 圖像轉換為 pdf
with open('merged.pdf', 'wb') as f:
    f.write(img2pdf.convert([open(i,"rb").read() for i in jpg_files]))
```

**Comment by chia-kai-chang** [comment]
123

**Reply by chia-kai-chang** [reply]
456

**Comment by Jinyuu17** [comment]
這個函式庫非常有用，如果圖片有重要的機密訊息，上傳到雲端轉檔是個非常不安全的舉動，難保圖片內容不被外洩。

**Comment by Jing-Asteroid** [comment]
感謝推薦，其實也有網頁版的線上轉檔器，但就是會有資料外洩的疑慮，所以每次再轉檔的時候都會很緊張。
用這個方式真的會放心很多，如果再寫一個迴圈包起來的話，應該就可以一口氣轉檔大量多維度的圖片資料集。

**Comment by zhilin93517** [comment]
我以前也有寫過類似東西，但是是用 OpenCV 把影片讀進來，轉檔之後再輸出，這樣可以比上傳到雲端更有隱私也有效率

**Comment by 112121py** [comment]
感謝版主推薦，之前也有類似的困擾，朋友有在網路上看到別人做的捷徑，設定完成也能達到同樣的效果，不過安全性就不太可判斷（？
